variadic_templates.cc

```diff
@@ -1,34 +1,57 @@
 /**
-* Example for section @section @title
-* of the C++ Booklet (https://goo.gl/VJ4T3A)
-* published by bvv software services AG (c) 2017
-*
-* This example illustrates the usage of variadic templates
-**/
+ * Example for section @section @title
+ * of the C++ Booklet (https://goo.gl/VJ4T3A)
+ * published by bvv software services AG (c) 2017
+ *
+ * This example illustrates the usage of variadic templates
+ **/
 
+#include <algorithm>
 #include <iostream>
+#include <memory>
 
 struct Dummy {
   int x;
   char c;
 };
 
 // variadic template function definition
-template <typename... Args> void aFunction(Args... args) {
+template <typename... Args> void aFunction(Args... args) {}
 
+// variadic template class definition
+template <typename... Args> class aClass {};
+
+std::string reorder_and_concat() { return std::string(); } // End of recursion
+
+// recursion popping the first template argument and concatenating
+template <typename T, typename... Args>
+std::string reorder_and_concat(T t, Args... args) {
+  std::string to_be_sorted(t);
+  std::sort(to_be_sorted.begin(), to_be_sorted.end());
+  return to_be_sorted + reorder_and_concat(args...);
 }
 
-// variadic template class definition
-template <typename... Args> class aClass {
+// using fold expression for simple operator +
+/*
+ * Supported operators are
+ * - boolean (&&, ||, <, >, ==, !=)
+ * - mathematical (+, -, *, /, %)
+ * - bitwise (&, |, ^, >>, <<)
+ * - memory (de-)referencing and obiect concatenation (.*, ->*)
+ * - assignement = and combinations such as /=, &=...
+ * - unpacking (,) [can be used to forward to normals functions]
+ */
+template <typename... Args> auto fold_using_add(Args &&... args) {

```

Comment:
Added example, I'm not too familiar with the fold expressions, @meshell  please confirm if I got the comments and the left/right correct.  
I updated the text in the booklet, but will add the example after your confirmation.